lib/matplotlib/axes/_axes.py

```diff
@@ -3861,7 +3861,7 @@ def scatter(self, x, y, s=20, c=None, marker='o', cmap=None, norm=None,
         # favor of mapping, not rgb or rgba.
         try:
             c_array = np.asanyarray(c, dtype=float)
-            if c_array.size == x.size:
+            if c_array.size == x.size and c_array.shape[0] != 1:
```

Comment:
Is there anyway we can get a numpy-scaler in here as a valid input?  If so probably want to be a bit more paranoid and add `and carray.ndims`